### DISCUSSION TITLE
Maximum population coverage Vs Maximum locations coverage

### DISCUSSION THREAD

**Discussion by rajesvariparasa** [pysal/spaghetti#553] [Q&A] [answered]
Hi,
The problem I am looking at is, I have about 300 slum locations with varying population numbers for each, and about 150 health facilities spread across a city. Now for selecting new facilities for maximum coverage, is it possible by the model, to account for covering **maximum slum population** and NOT maximum number of slum locations? i.e. Is there a way to prioritize coverage of some locations over the others using population as weights to the location? How can I implement this?

Thank you!

**Comment by jGaboardi** [comment, marked as answer]
@rajesvariparasa Thanks for starting this discussion here. It seems like the [Maximal Covering Location Problem](https://pysal.org/spaghetti/references.html#church1974) would suit your needs. With the MCLP you can set the desired number of facilities ***p*** as a parameter, which the model will then use determined maximum weighted coverage (population in your case. I provide an example of this [here](https://pysal.org/spaghetti/notebooks/facility-location.html#3.4-Maximal-Covering-Location-Problem) in the `spaghetti` tutorials where I use randomly-generated points for the facilities (Little Free libraries) and clients (synthetic households) along with a `weight` parameter (artificial population), which is described [here](https://pysal.org/spaghetti/notebooks/facility-location.html#1.1-Scenario:).

* Richard L. Church and C.S. ReVelle. The Maximal Covering Location Problem. Papers in Regional Science Association, 32:101–118, 1974. doi:doi.org/10.1111/j.1435-5597.1974.tb00902.x.
